backend.go

```diff
@@ -164,11 +174,22 @@ func (b *kubeAuthBackend) loadConfig(ctx context.Context, s logical.Storage) (*k
 	}
 
 	// Read local CA cert unless it was stored in config.
+	// Else build the TLSConfig with the trusted CA cert and load into client
 	if config.CACert == "" {
 		config.CACert, err = b.localCACertReader.ReadFile()
 		if err != nil {
 			return nil, err
 		}
+	} else {
+		certPool := x509.NewCertPool()
+		certPool.AppendCertsFromPEM([]byte(config.CACert))
+
+		tlsConfig := &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			RootCAs:    certPool,
+		}
+
+		b.httpClient.Transport.(*http.Transport).TLSClientConfig = tlsConfig
```

Comment:
There might be two issues if we set this on `loadConfig`:

1. We'd have to grab a write lock during this portion of the call since we're modifying the backend's HTTP client. However, `loadConfig` acts more like a "read" operation that retrieves the config from storage.
2. We'd get to a state where we're now setting/updating the client on every `loadConfig` call, which happens on every login (and aliasLookahead) call.

We could get around issue #2 by using a sentinel value to track whether we've set this already, and only set it once during the lifecycle of the backend object. However, it might be simpler to set it when the backend is instantiated (i.e the `Backend()`) or within a backend's `InitializeFunc`.

Reply:
@tomhjp I also wanted to loop you in on [d1e904b](https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/142/commits/d1e904bd7f5f955105a51bf0d9464c5c52ca888a) I added an `initialize` method to the `kubeAuthBackend` struct. The reasoning for this is because we don't have access to the config without creating additional logic inside the `Backend()` func which is primarily just used for setup/configuration it made more sense to try and load the config on initialization when the auth backend is mounted. This still allows us to continue as normal if a config doesn't exist and if a config and a trusted CA cert exist we can load it into the HTTP client. @calvn showed me a great example of this flow being used in the AWS auth method [here](https://github.com/hashicorp/vault/blob/86c1cdf9e61897dee14203c51d6ad3d971cfda1a/builtin/credential/aws/backend.go#L189).